stdlib/date/date.go

```diff
@@ -147,6 +146,29 @@ func init() {
 				return nil, fmt.Errorf("cannot convert argument t of type %v to time", v1.Type().Nature())
 			}, false,
 		),
+		"truncate": values.NewFunction(
+			"truncate",
+			semantic.NewFunctionPolyType(semantic.FunctionPolySignature{
+				Parameters: map[string]semantic.PolyType{"t": semantic.Time, "unit": semantic.Duration},
+				Required:   semantic.LabelSet{"t", "unit"},
+				Return:     semantic.Time,
+			}),
+			func(args values.Object) (values.Value, error) {
+				v, ok := args.Get("t")
+				u, unitOk := args.Get("unit")
+				if !ok {
```

Comment:
Can you move this directly below line 157 and directly use `GetRequiredTime`/`Duration`. That would simplify logic.

Reply:
`args.GetRequiredTime` and `args.GetDuration` don't pop up when I type it in?

Reply:
It's here: https://github.com/influxdata/flux/blob/c7e454395264dcf1cb23f9889ef7bbbaa947fc4c/compile.go#L682

Reply:
I think `args` is `*TableObject` not `Arguments`, so how would that work?

Reply:
Oh I see, sorry about that! This code is ok though :)